Add short page on event data (#156)
axiomhq/docs: getting-started-guide/event-data.mdx

---
title: "Event data"
description: "This page explains the fundamentals of timestamped event data in Axiom."
sidebarTitle: Event data
---

> Axioms’s mission is to help you leverage the power of timestamped event data.

Reviewer: ```suggestion
Axiom’s mission is to operationalize every bit of event data in your organization.
```

Neil provided this specific definition of our mission earlier in the year, and I think it fits well here. The next paragraph then uses the longer form of timestamped event data, which feels ok to me.

> Axiom considers timestamped event data to be the record of every digital interaction between human, sensor, and machine. Event data is the atomic unit of activity. For this reason, every function in any business with digital activity can benefit from leveraging event data.

Reviewer: ```suggestion
Timestamped event data records every digital interaction between human, sensor, and machine, making it the atomic unit of activity for organizations. For this reason, every function in any business with digital activity can benefit from leveraging event data.
```

Do we need to scope this to Axiom’s belief system, or hold it as truth? This reads with more conviction to me.

Each event is simply a structured record—composed of key-value pairs—that captures meaningful interactions or changes in state within a system. While these can appear in various forms, they usually contain the following:

- **Timestamp**: When the event occurred.
- **Attributes**: A set of key-value pairs offering details about the event context.

> - **Metadata**: Contextual labels and IDs that connect related events.

Reviewer: I’m not totally sure there is a distinction that needs to be drawn between attributes and metadata; I would probably use them interchangeably, but it doesn’t offend me.

<Note>

> The timestamped event data that Axiom works with is not to be confused with time-series event data.

Reviewer: Without more context, I would omit this note. It doesn’t help the reader much imo. 

Usually, when using this kind of language, we’re creating a distinction between time series databases (like Timescale), and Axiom. 

I might draw on Neil’s idea that Axiom is suited to answering questions across operational (now, individual), analytical (recent history, population) and archival workloads, where time series databases might not have quite the same applied nature. But others at Axiom might be able to add more.

</Note>

## Uses of event data

> Event data, understood as the atomic unit of digital activity, is the lifeblood of modern businesses. Leveraging the power of event data is essential in the following areas, among others:

Reviewer: Ok, this is a neat way to hone in on some specific use cases. 👍

- [Observability](/getting-started-guide/observability)
- Security
- Business intelligence
- AI and machine learning

> - Finance

Reviewer: Can we remove Finance and put Product analytics above business intelligence? That feels like a more natural selection and sequencing today. Sorry, not sure I can do multi-line suggestions inline in GitHub.

---
title: "Events in Axiom"
description: "This page explains the fundamentals of event data in Axiom."
sidebarTitle: Events
---

> In Axiom, event data is the atomic unit of activity, encompassing logs, traces, and even product interactions. Each event is simply a structured record—composed of key-value pairs—that captures meaningful interactions or changes in state within a system. While these can appear in various forms, from log lines to complex traces, they usually contain the following:

Reviewer: Let’s call it "timestamped event data" in the first mention, and event data thereafter. I want to ground event data very clearly in time from the outset. Note: This is _not_ described as time _series_ event data. The language choice is intentional, to distance Axiom rightly from time series DBs.

> In Axiom, event data is the atomic unit of activity, encompassing logs, traces, and even product interactions. Each event is simply a structured record—composed of key-value pairs—that captures meaningful interactions or changes in state within a system. While these can appear in various forms, from log lines to complex traces, they usually contain the following:

Reviewer: In the past, I have said that event data is the record of every interaction between human, sensor, and machine. i can’t remember where that originated from, either cfrln or a deep dive on the history of event data ~Q1/Q2, but the reason I gravitated towards it is because it zooms us out from any use case first, to create an appreciation for the total scope and opportunity before honing in. 

By putting logs and traces as the first and second example, it feels to me that we have already coupled Axiom so tightly to engineering use cases, and then we try and walk it back with the third example, product interactions. I really want the reader to feel that timestamped event data is _the_ record of absolutely every digital interactional - the digital breadcrumb - and _then_ (and only then) for us to contextualize that with domain-specific examples. That could work better as two sentences.

> In Axiom, event data is the atomic unit of activity, encompassing logs, traces, and even product interactions. Each event is simply a structured record—composed of key-value pairs—that captures meaningful interactions or changes in state within a system. While these can appear in various forms, from log lines to complex traces, they usually contain the following:

Reviewer: > While these can appear in various forms, from log lines to complex traces

Again, I feel the same way as above. This innocuous "from" scopes the breadth of use cases to engineering when I read this. The framing is changed from what I want the world to know, which is that every function in any business with digital activity can and will benefit from Axiom.

- **Timestamp**: When the event occurred.
- **Attributes**: A set of key-value pairs offering details about the event context.

> - **Metadata**: Contextual labels and IDs, such as trace ID and span ID, that connect related events.

Reviewer: For the avoidance of doubt, I feel the same way here about "such as trace ID and span ID".

> Rather than simply calling them logs or metrics, event data reflects a broader range of interactions, crossing boundaries from engineering to product management, security, and beyond. This includes everything from the number of documents created in a product to the sequence of API calls during a checkout process.

Reviewer: Good. 

"Whole engineers might most often work with timestamped event data in the form of logs or metrics..."

Something like this ^ might be more effective. I don’t love the phrasing, but the goal again would be to say: We know our most common reader is likely to be an engineer because of Axiom’s history, but that there is a bigger opp here for a business.

> ## Types of event data

Reviewer: I know that you want to set up the punchline of Axiom’s metrics compatibility, but the problem here is we have now gone from what I hope will be a world view that timestamped event data shows up everywhere, to narrowing down even _more_ than engineering to observability.

If I look at this page, I would definitely leave thinking Axiom is just another engineering-focused tool. How could we change that perception? 

Does it need a heading that shows we are narrowing to the engineering use case specifically? Is this section or mention of observability needed at all?

Traditionally, observability has been associated with three pillars, each effectively a specialized view of event data:

- **Logs**: Logs record discrete events, such as error messages or access requests, typically associated with engineering or security.
- **Traces**: Traces track the path of requests through a system, capturing each step’s duration. By linking related spans within a trace, developers can identify bottlenecks and dependencies.
- **Metrics**: Metrics quantify state over time, recording data like CPU usage or user count at intervals. Product or engineering teams can then monitor and aggregate these values for performance insights.

> In Axiom, these observability elements are stored as event data, allowing for fine-grained, efficient tracking across all three pillars.

Reviewer: "Just like any other type of (timestamped) event data, ..."

Author: So far mistakenly considered not event data, but in fact also event data

> ## Metrics support

Reviewer: Putting all of the above to one side, yes, this is good. It answers the question of our current metrics status, which is not well covered elsewhere.
